libraries/abstractions/common_io/test/test_iot_timer.c

```diff
@@ -274,44 +270,45 @@ TEST( TEST_IOT_TIMER, AFWP_IotTimer_Callback )
     int32_t lRetVal;
 
     /* Open timer to initialize hardware. */
-    if( gIotTimerHandle == NULL )
-    {
-        xTimerHandle = iot_timer_open( ltestIotTimerInstance );
-        TEST_ASSERT_NOT_EQUAL( NULL, xTimerHandle );
-    }
-    else
-    {
-        xTimerHandle = gIotTimerHandle;
-    }
-
-    /* Set up the callback */
-    iot_timer_set_callback( xTimerHandle,
-                            prvTimerCallback,
-                            NULL );
-
-    /* Set up the delay */
-    lRetVal = iot_timer_delay( xTimerHandle,
-                               testIotTIMER_DEFAULT_DELAY_US );
-    TEST_ASSERT_EQUAL( IOT_TIMER_SUCCESS, lRetVal );
+    xTimerHandle = iot_timer_open( ltestIotTimerInstance );
+    TEST_ASSERT_NOT_EQUAL( NULL, xTimerHandle );
 
-    /* Start the timer */
-    lRetVal = iot_timer_start( xTimerHandle );
-    TEST_ASSERT_EQUAL( IOT_TIMER_SUCCESS, lRetVal );
+    if( TEST_PROTECT() )
+    {
+        /* Set up the callback */
+        iot_timer_set_callback( xTimerHandle,
+                                prvTimerCallback,
+                                NULL );
+
+        /* Set up the delay */
+        lRetVal = iot_timer_delay( xTimerHandle,
+                                   testIotTIMER_DEFAULT_DELAY_US );
+
+        /* Expect either "Not running" or "Success" */
+        if( lRetVal != IOT_TIMER_NOT_RUNNING )
```

Comment:
You need to add the iot_timer_delay after starting the timer so that semaphoreTake can be unblocked. Since we are updating the test to check for NOT_RUNNING, lets also fix the test to do the delay after starting the timer and expect the semaphore take to succeed.

Reply:
@vjjunnut II have asked this question to @lundinc2 before to what the expected behavior for calling a delay after start is. My understanding is that setting a delay before start will still result in the delay being perform from the point of "start" which means the semaphore will be unblocked as we do start the timer. 

The expected behavior of "delay" being called after the timer has been started is unclear. I would imagine that it would start the "delay counter" right away but this was not what the tests implied at first which is why I have not altered the test sequence, just added a check for possible return values as these are clearly defined in the API.

Reply:
The test had a bug. We can sync up internally and get back.